fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -569,14 +692,32 @@ def upload(app_id = nil, dir = nil, package_path: nil, asset_path: nil)
       password_placeholder = @jwt.nil? ? 'YourPassword' : nil
       jwt_placeholder = @jwt.nil? ? nil : 'YourJWT'
 
-      command = @transporter_executor.build_upload_command(@user, @password, actual_dir, @provider_short_name, @jwt)
-      UI.verbose(@transporter_executor.build_upload_command(@user, password_placeholder, actual_dir, @provider_short_name, jwt_placeholder))
+      # Handle AppStore Connect API
+      use_api_key = !@api_key.nil?
+      api_key_placeholder = nil unless use_api_key
+      api_key_placeholder = { key_id: "YourKeyID", issuer_id: "YourIssuerID" } if use_api_key
+
+      if use_api_key
+        # Specified p8 needs to be generated to call altool
+        tmp_p8_file_path = File.join(Dir.tmpdir, "AuthKey_#{@api_key[:key_id]}.p8")
+        p8 = File.open(tmp_p8_file_path, "w")
+        p8.write(@api_key[:key])
+        p8.close
```

Comment:
**Question:** **if the API-key is Base64 encoded** then i think, we need to decode it first before writing it into the file? 

- Example: [Base64 API-Key example-link](https://github.com/fastlane/fastlane/blob/master/spaceship/spec/connect_api/fixtures/asc_key_base64.json)
- Please see this param https://github.com/fastlane/fastlane/blob/78488dddef57c9d6afbd9a6c04967dcc9c61858b/fastlane/lib/fastlane/actions/app_store_connect_api_key.rb#L74-L76
- i think, we need to add similar logic like below https://github.com/fastlane/fastlane/blob/ff41f3556f1baf8e700477e50f65be728780aa94/spaceship/lib/spaceship/connect_api/token.rb#L64-L66

Reply:
Ah, it is true, we need to convert it from base64, let me fix soon

Reply:
https://github.com/fastlane/fastlane/pull/20631/commits/4f677a6e6e99134ee6ca603b85d2dac582e4a1be Please check it and test just for case 🙇